### PR TITLE
Use CPack to make a new DMG

### LINKED ISSUE #94: Redo macos distribution packaging
There was discussion a while back about redoing the dmz/zip/app bundle distribution for MacOS and some initial work for it. This was put on the back burner for a variety of reasons but we should re-visit it now that signing is done and we are switching to cmake.

Currently we rely on qmake to produce an app bundle, qtmacdeploy to make the app bundle distributable (providing Qt and updating paths), and do signing separately. The app bundle is stored in a dmg because GitHub actions can't upload a directory to store files between steps and we need to preserve permissions. This all then gets zipped with the README and licensing info because it's easier than adding to the dmg. This requires first unzipping then mounting the dmg to install which is less than ideal.

Moving forward we should create a single dmg, which is the distribution artifact, that includes the app bundle, README, license, and any other relevant distribution info and create a nice background image with links to the `Applications` directory for easy installation.

This should probably be done after #72 and #93 are completed to avoid having to go through the effort multiple times. This will likely require completely moving away from qtmacdeploy as it doesn't seem configurable enough and probably rewriting some of the signing steps for the new workflow.

### PR DESCRIPTION
**Pull request by crizzitello**
Based on #121 
Fixes: #94 
 - CPack will make now make a DMG file
 - Signing happens with a new CMake Option "NOTARIZE_AS"
 - DMG Will mount with custom icon drive_icon
 - DMG has Background image
window

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -44,25 +44,39 @@ jobs:
...
             , linuxDeployQtPath: "export PATH=$PATH:/opt/Qt/6.2.4/gcc_64/libexec"
-
           }
         - {
             name: "Mac-Qt5"
             , os: macos-latest
             , QT_VERSION: 5.15.2 , QT_INST_DIR: /Users/runner
             , QT_STRING: "Qt5"
-            , extraCMakeConfig: "-DQT_DEFAULT_MAJOR_VERSION=5"
+            , extraCMakeConfig: "-DQT_DEFAULT_MAJOR_VERSION=5 -DNOTARIZE_AS=\"John Kennedy\""
```
Comment (JoelAtDeluxe) [inline .github/workflows/ci.yaml:71]: Should we notarizing as `John Kenendy`, or as `The Paranoids`, or something similar? Not to deflate John's importance, but I just wonder if we want to tie this to a person, or an organization.

Comment (crizzitello) [reply .github/workflows/ci.yaml:71]: Yeah since its an org I would use the org name. Would if anything just look more uniform if anyone checks the signer

Comment (jkennedyvz) [reply .github/workflows/ci.yaml:71]: We notarize as my ID for the time being - there is no legal entity/Apple account for The Paranoids.

Comment (crizzitello) [reply .github/workflows/ci.yaml:71]: I figured that was the reason. If you want to change that in the future it can easily be changed.

**Comment by jrozner** [comment]
For the DMG background can you provide size requirements? I'd like to get something a bit more branded

**Comment by crizzitello** [comment]
> For the DMG background can you provide size requirements? I'd like to get something a bit more branded
There is no size requirement; there are only two requirements 
  - It must be 72 DPI
  - It must be a png file.
 The current image is 800x600@72dpi

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -170,78 +154,68 @@ jobs:
...
+            fi
+        fi
+        mv ashirt-*.* dist/
 
     - name: Archive production artifacts
       uses: actions/upload-artifact@v2
       with:
-        name: ${{ env.name }}
-        path: dist
+        name: ashrit-${{env.githash}}
```
Comment (jkennedyvz) [inline .github/workflows/ci.yaml]: ^ typo changing the output name

Comment (crizzitello) [reply .github/workflows/ci.yaml]: Was in #121  (now fixed in both)

**Review by jkennedyvz — APPROVED** [review]

**Pull request merged**